### PR TITLE
LUM-23665 Add y2k_birthday update by query

### PR DESCRIPTION
**Pull request by Blackbaud-ClareColeman**
- Add support for slicing
- Add script for a long-running update by query

@blackbaud/team-cerebro @Blackbaud-RajRaulji

### CODE REVIEW

File: indexUtil.sh [hunk 1]
```diff
@@ -127,6 +131,7 @@ function buildUpdateByQueryUrl() {
     local _env=$2
     local _idxName=$3
     local _docType=$4
+    local _slices=$5
```
Comment (Blackbaud-RajRaulji) [inline indexUtil.sh:134]: where are you using _slices ?

File: scripts/y2k_birthday_update_query.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+    "query": {
+        "exists": { "field": "date_of_birth" }
```
Comment (Blackbaud-RajRaulji) [inline scripts/y2k_birthday_update_query.json]: will the "exists" filter, filter out records will null value for "date_of_birth".

File: indexUtil.sh (hunk 1 above)
Comment (Blackbaud-ClareColeman) [reply indexUtil.sh:134]: `updateByQueryLongRunning.sh` and only there. I noted that in the README, and also noted it was required in the comment header on that file.

File: scripts/y2k_birthday_update_query.json (hunk 2 above)
Comment (Blackbaud-ClareColeman) [reply scripts/y2k_birthday_update_query.json]: Yep! And `[]` and `[null]` and... :)

**Comment by Blackbaud-ClareColeman** [comment]
@blackbaud/team-cerebro I've updated the script to ignore version conflicts and documents we've already touched with this script.

**Review by Blackbaud-ChrisMecklenborg — APPROVED** [review]
+1

**Pull request merged**